### PR TITLE
V2 support

### PR DESCRIPTION
**Pull request by thelostone-mc**

### CODE REVIEW

File: contracts/MultiRoundCheckoutAllocate.sol [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        ) {
+            revert ExcessAmountSent();
+        }
+    }
+
+    /**
+     * allocateDAIPermit: _data for multiple rounds at once with DAI.
+     */
+    function allocateDAIPermit(
+        address allo,
```
Comment (thelostone-mc) [inline contracts/MultiRoundCheckoutAllocate.sol]: we could set this as a storage variable

File: contracts/MultiRoundCheckout.sol [hunk 2]
```diff
@@ -8,16 +8,35 @@ import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/IERC20PermitU
...
+    }
+
+    modifier validateErc20Balance(address token) {
+        uint256 initialBalance = IERC20Upgradeable(token).balanceOf(address(this));
+        _;
+        if (IERC20Upgradeable(token).balanceOf(address(this)) != initialBalance) {
+            revert ExcessAmountSent();
+        }
+    }
+
```
Comment (gravityblast) [inline contracts/MultiRoundCheckout.sol:34]: shall we add the constructor with `_disableInitializers();` too as proposed by @0xKurt ?

File: contracts/MultiRoundCheckout.sol [hunk 3]
```diff
@@ -8,16 +8,39 @@ import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/IERC20PermitU
...
+    constructor() {
+        _disableInitializers();
```
Comment (0xKurt) [inline contracts/MultiRoundCheckout.sol:37]: you may also add `_pause` ?

Comment (gravityblast) [reply contracts/MultiRoundCheckout.sol:37]: good idea, done!

**Pull request closed without merging**